arduino-ide-extension/src/browser/serial/monitor/monitor-widget.tsx

```diff
@@ -156,11 +174,23 @@ export class MonitorWidget extends ReactWidget {
     ];
   }
 
-  private getCurrentSettings(): Promise<MonitorSettings> {
+  private async startMonitor(): Promise<void> {
+    await this.appStateService.reachedState('ready');
```

Comment:
Thanks!

This would imply that the monitor widget knows how the board service provider behaves, which I did not want. If we change how the board service provider starts, we would also have to change the monitor widget implementation, which would be unrelated. That's why both the monitor and the board service provider wait for the app `ready` event.

Reply:
Makes sense, I didn't think about that 👍